src/elements/play-assets/play-assets.ts

```diff
@@ -0,0 +1,396 @@
+import {type PropertyValues, ReactiveElement} from 'lit'
+import {customElement, property} from 'lit/decorators.js'
+import {WebAccess, WebStorage} from '@zenfs/dom'
+import * as ZenFS from '@zenfs/core'
+import {type FileSystem, fs, InMemory} from '@zenfs/core'
+import * as IDB from 'idb-keyval'
+import {
+  hasFileAccessAPI,
+  tryGetFile,
+  tryQueryPermission
+} from '../../utils/file-access-api.js'
+import {Zip} from '@zenfs/zip'
+import {Bubble} from '../../utils/bubble.js'
+
+declare global {
+  interface HTMLElementEventMap {
```

Comment:
please move these events to the elements that actually emit them. I know the typing all coalesces in the interface but there's no way I'm going to think to update or read the event here if I'm in the element actually doing the emitted.